src/GuzzleHttp/Cache/MockStorage.php

```diff
@@ -0,0 +1,131 @@
+<?php
+
+/*
+ * This file is part of the CsaGuzzleBundle package
+ *
+ * (c) Charles Sarrazin <charles@sarraz.in>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code
+ */
+
+namespace Csa\Bundle\GuzzleBundle\GuzzleHttp\Cache;
+
+use GuzzleHttp\Psr7\Response;
+use Psr\Http\Message\RequestInterface;
+use Psr\Http\Message\ResponseInterface;
+
+class MockStorage implements StorageAdapterInterface
+{
+    /**
+     * @var string
+     */
+    private $storagePath;
+
+    /**
+     * @var array
+     */
+    private $headersBlacklist;
+
+    /**
+     * @param $storagePath
+     *
+     * @param null|array $headersBlacklist
+     */
+    public function __construct($storagePath, $headersBlacklist = null)
+    {
+        $this->storagePath = $storagePath;
+
+        if (is_array($headersBlacklist)) {
+            $this->headersBlacklist = $headersBlacklist;
+        } else {
+            $this->headersBlacklist = [
+                'User-Agent',
+                'Host',
+            ];
+        }
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function fetch(RequestInterface $request)
+    {
+        $path= $this->getPath($request);
+
+        if (!file_exists($path)) {
+            throw new \RuntimeException('Record not found.');
+        }
+
+        $data = unserialize(file_get_contents($path));
+
+        return new Response($data['status'], $data['headers'], $data['body'], $data['version'], $data['reason']);
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function save(RequestInterface $request, ResponseInterface $response)
+    {
+        $path = $this->getPath($request);
+
+        $requestData = [
+            'method'  => $request->getMethod(),
+            'uri'     => $request->getUri(),
+            'headers' => $request->getHeaders(),
+        ];
+
+        $responseData = [
+            'status' => $response->getStatusCode(),
+            'headers' => $response->getHeaders(),
+            'body' => (string)$response->getBody(),
+            'version' => $response->getProtocolVersion(),
+            'reason' => $response->getReasonPhrase(),
+        ];
+
+        file_put_contents($path, serialize($responseData));
+
+        // Save in json for debug
+        file_put_contents($path.'.json', json_encode($requestData, JSON_PRETTY_PRINT)."\n\n".json_encode($responseData, JSON_PRETTY_PRINT));
```

Comment:
Great idea ! 
But we need something to dump/load from a PSR-7 response. Do you know an existing class to do this ?
If not, I will write my own one.